usd_qtpy/prim_hierarchy.py

```diff
@@ -410,38 +399,113 @@ def create_prim(action):
+                    "Clear the default prim from the stage's root layer.\n"
+                    f"The current default prim is {default_prim.GetName()}"
+                )
+                action.setToolTip(tip)
+                action.setStatusTip(tip)
+                action.triggered.connect(partial(stage.ClearDefaultPrim))
+            else:
+                label = "Set as default prim"
+                action = menu.addAction(label)
+                tip = "Set prim as default prim on the stage's root layer."
+                action.setToolTip(tip)
+                action.setStatusTip(tip)
+                action.triggered.connect(partial(stage.SetDefaultPrim, parent))
+
+        # Allow referencing / payloads / variants management
+        if parent != root:
+
+            def _add_reference(prim, as_payload=False):
+                filenames, _filter = QtWidgets.QFileDialog.getOpenFileNames(
+                    parent=self,
+                    caption="Sublayer USD file",
+                    filter="USD (*.usd *.usda *.usdc);"
+                )
+                if not filenames:
+                    return
 
-        menu.triggered.connect(create_prim)
+                if as_payload:
+                    payloads = prim.GetPayloads()
+                    for filename in filenames:
+                        payloads.AddPayload(filename)
+                else:
+                    references = prim.GetReferences()
+                    for filename in filenames:
+                        references.AddReference(filename)
+
+            action = menu.addAction("Add reference")
+            action.triggered.connect(partial(_add_reference, parent, False))
+            action = menu.addAction("Add payload")
+            action.triggered.connect(partial(_add_reference, parent, True))
+
+            def _add_variant_set(prim):
+                # Prompt for a variant set name (and maybe directly allow
+                # managing the individual variants from the same UI; and allow
+                # picking the default variant?)
+                prompt = CreateVariantSetDialog(parent=self)
+                if prompt.exec_() == QtWidgets.QDialog.Accepted:
+                    name = prompt.name.text()
+                    if name:
+                        # Create the variant set, even allowing to create it
+                        # without populating a variant name
+                        prim.GetVariantSets().AddVariantSet(name)
+
+            action = menu.addAction("Create Variant Set")
+            action.triggered.connect(partial(_add_variant_set, parent))
 
         # Get mouse position
         global_pos = self.viewport().mapToGlobal(point)
         menu.exec_(global_pos)
 
+    def on_prim_tag_clicked(self, index, text):
+        print(index.data())
+        if text == "DFT":
+            print("DFT YES")
+        elif text == "REF":
+            print("REF YES")
+        elif text == "VAR":
+            print("VAR YES")
```

Comment:
Clean up this debug code - we should implement something useful on clicking these - e.g. allow authoring variants in more detail